lib/EndOfItem/tests/EndOfItem.test.js

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+
+import '../../../test/jest/__mock__';
+
+import { renderWithIntl } from '../../../test/jest/helpers';
+import { EndOfItem } from '../EndOfItem';
+
+const setupEndOfItem = ({
+  title,
+  className,
+} = {}) => renderWithIntl(
+  <EndOfItem
+    title={title}
+    className={className}
+  />,
+  [{
+    prefix: 'stripes-components',
+    translations: { endOfList: 'stripes-components.endOfList' },
+  }]
+);
+
+describe('EndOfItem', () => {
+  const customClass = 'my-class';
+
+  describe('rendering EndOfItem with title', () => {
+    const title = 'Title';
+
+    it('should add custom class style to layout container', () => {
+      setupEndOfItem({
+        title,
+        className: customClass,
+      });
+      const layout = document.querySelector('[data-test-layout]');
+
+      expect(layout.classList.contains(customClass)).toBeTruthy();
+    });
+
+    it('should display "end mark" svg icon', () => {
+      setupEndOfItem({ title });
+      const icon = document.querySelector('[data-test-icon-element]');
+      const hasIconEndMarkClass = icon.classList.contains('icon-end-mark');
+
+      expect(icon).toBeInTheDocument();
+      expect(hasIconEndMarkClass).toBeTruthy();
+    });
+
+    it('should display passed title', () => {
+      setupEndOfItem({ title });
+      const label = document.querySelector('span.label');
+
+      expect(label.innerHTML).toEqual(title);
+    });
+  });
+
+  describe('rendering EndOfItem without properties', () => {
+    it('should not add custom class style to layout container', () => {
+      setupEndOfItem();
+      const layout = document.querySelector('[data-test-layout]');
+
+      expect(layout.classList.contains(customClass)).toBeFalsy();
```

Comment:
I think it's better to check, that layout only contains default classes, rather than checking that it does not contain some class because you can't be sure, that there's no other extra class

Reply:
I don't agree on this and see that we already do such in the preloader. I think otherwise it is better to check for the custom class only in order to prevent other classes be present in tests which are really implementation details.

Reply:
Agreed to remove such tests at all as there is hard to imagine such tests to fail as the class is not going to magically appear if it not somehow provided.